feat: add setting to change shape colors

tom-anders opened the pull request:
Closes #841

[shapeColors.webm](https://github.com/user-attachments/assets/84ab5eef-5e28-442b-b97e-50657201a3a8)

tom-anders commented on lib/src/model/settings/board_preferences.dart:
```diff
@@ -137,6 +148,23 @@ class BoardPrefs with _$BoardPrefs {
+  purple;
+
+  Color get color => switch (this) {
+        ShapeColor.green => LichessColors.green,
```
Are these the correct colors to use here? Or should we rather use the colors from here: https://github.com/lichess-org/chessground/blob/54a7e71bf88701c1109d3b9b8106b464012b94cf/src/state.ts#L178

veloce replied:
I'd say let's use the colors defined in lichess website indeed.

Also I think showing the name of the color in the picker is not enough. Could you also show a coloured square so the user can look at the color? Thanks.

tom-anders replied:
> I'd say let's use the colors defined in lichess website indeed.
👍 

> Also I think showing the name of the color in the picker is not enough. Could you also show a coloured square so the user can look at the color? Thanks.

Ah good idea! Maybe we could even show a small arrow instead of a square?

tom-anders commented:
Ok, made the following changes:

1) Using same colors as lichess.org now. IIRC blue, green, red and blue are the arrow colors supported on the website
2) Moved the setting to the new "Theme" screen, this way we can add the shapes to the preview board.
3) Added a colored box to the chouice picker entries, although arguably this is now not needed anymore, as the user already gets visual feedback on the preview board.

[shapeColors.webm](https://github.com/user-attachments/assets/d049f081-cd3d-4f59-8010-48cdaba562be)

veloce commented:
Great idea to put it in the theme screen! There are some weird conflicts in the game history screen @tom-anders can you see that please?

veloce commented:
> 3. Added a colored box to the choice picker entries, although arguably this is now not needed anymore, as the user already gets visual feedback on the preview board.

It is still useful for iOS users because the picker will be on a different screen. 

Another note: I don't think we should return the `/ blue` color in the settings value preview because:
1. it is not that clear this is about the color of the arrow
2. space is constrained on iOS devices and this should be kept short

tom-anders commented:
> Another note: I don't think we should return the `/ blue` color in the settings value preview because:

Done!

Also, fixed conflicts now.
